src/core/connection.c

```diff
@@ -3664,6 +3661,9 @@ QuicConnRecvPrepareDecrypt(
                 return FALSE;
             }
             Packet->KeyType = QUIC_PACKET_KEY_1_RTT_NEW;
+        } else {
+            QuicPacketLogDrop(Connection, Packet, "This packet has already been read");
```

Comment:
Is this the right error to send here. Or should we let code later down the path handle this case?

Reply:
This only happens when the packet has the same packet number as the ReadKeyPhaseEndPacket, right? If it's a duplicate, then it should be handled elsewhere with the other duplicate packets.

Reply:
It happens if its anywhere in the range between start and end. From here, it would basically go to decryption, and either fail to decrypt, or decrypt and be detected as a duplicate packet.

Reply:
In the packet reordering scenario, this seems like it could cause valid packets to be dropped, I think it might be best to do nothing in this case and let it get handled elsewhere.

Reply:
This only happens if a packet in the opposite key phase is received that is in the range of packets from the current key phase. So in theory it should never be hit, even with reordering. If I send it through, should it be an old key or the current key?

Reply:
Turns out this case is actually hit in stateless reset. So we just need to let it fly.